scripts/sorting.py

```
@@ -59,20 +61,37 @@ def _run_sorting_job(algorithm: str, recording_nwb_uri: str, sorting_params: dic
             'sorting_npz_uri': sorting_npz_uri
         }
 
+def _run_sorting_jobs_wrapper(job: Job, **kwargs):
+    print(f'Running: {job.label}')
+    output = _run_sorting_job(**job.kwargs, **kwargs)
+    print(f'OUTPUT of {job.label}:\n{output}')
+    kc.set(job.key(), output)
+
+
 @click.command()
 @click.argument('config_file')
 @click.argument('algorithm')
-@click.option('--force-run', is_flag=True, help="Force rerun")
+@click.option('--max-simultaneous-sorts', help="Maximum number of sorting jobs to run simultaneously")
+@click.option('--force-run', is_flag=True, help="Force rerurn")
 @click.option('--rerun-failing', is_flag=True, help="Rerun the failing jobs")
 @click.option('--docker', is_flag=True, help="Use docker image")
 @click.option('--singularity', is_flag=True, help="Use singularity image")
 @click.option('--image', default=None, help='Image for use in docker or singularity mode')
-def main(config_file: str, algorithm: str, force_run: bool, rerun_failing: bool, docker: bool, singularity: bool, image: Union[str, None]):
+def main(config_file: str, algorithm: str, max_simultaneous_sorts: Union[int, None], force_run: bool, rerun_failing: bool, docker: bool, singularity: bool, image: Union[str, None]):
+    if docker and singularity:
+        raise Exception('Both singularity and docker were requested, but no more than one can be used simultaneously.')
+    try:
+        max_simultaneous_sorts = int(max_simultaneous_sorts)
+        if max_simultaneous_sorts < 1:
+            max_simultaneous_sorts = 1
+    except:
+        max_simultaneous_sorts = 1
```

Not sure why you have try/except here. If there is an error in parsing the argument, we really should raise an exception. Otherwise the parameter may silently be set to the default if something unexpected goes wrong.

---

Was aiming for fault tolerance; but I'll remove the `try` so that this halts execution on bad input.

---

Ah, now I remember. The `try` statement handled the most-likely-case (user didn't set the value so it's None) without a separate check. No worries, I'll pull it.